proposals/eligibility_validator.go

```diff
@@ -140,11 +141,15 @@ func (v *Validator) validateReference(ballot *types.Ballot, owned *types.Activat
 	if ballot.EpochData.Beacon == types.EmptyBeacon {
 		return nil, fmt.Errorf("%w: ref ballot %v", errMissingBeacon, ballot.ID())
 	}
-	if len(ballot.ActiveSet) == 0 {
+	activeSet, err := activesets.Get(v.cdb, ballot.EpochData.ActiveSetHash)
```

Comment:
there should be a way to avoid this read. we should have loaded this data earlier, as we need to check data availability. and then we need to pass it together with the rest of the ballot for eligibility validation

Reply:
thanks. changed to pass it from the original lookup.